packages/fork-choice/src/forkChoice/forkChoice.ts

```diff
@@ -931,16 +931,32 @@ export class ForkChoice implements IForkChoice {
    * Return only the non-finalized blocks.
    */
   getAllAncestorBlocks(blockRoot: RootHex): ProtoBlock[] {
-    const blocks = this.protoArray.getAllAncestorNodes(blockRoot);
-    // the last node is the previous finalized one, it's there to check onBlock finalized checkpoint only.
-    return blocks.slice(0, blocks.length - 1);
+    const {ancestors} = this.protoArray.getAllAncestorAndNonAncestorNodes(blockRoot);
```

Comment:
but we call this here are we getting a performance regression on `getAllNonAncestorBlocks` (same for `getAllNonAncestorBlocks`), or is this different from the concerns raised by @nazarhussain and @matthewkeil earlier?